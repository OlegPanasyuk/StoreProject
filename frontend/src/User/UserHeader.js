import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Row, Col, Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';


//Redux
import { connect } from 'react-redux';
import { setUserInfo } from '../REDUX/actions/actionsUser';
import { showHistoryOfShoppping } from '../REDUX/actions/actionsHistoryOfShopping';
import {
    showUserHistory,
    showUserProfile,
    showCatalogue
} from '../REDUX/actions/actionsMainContent';

// For Requests to server
import rest from 'rest';
import pathPrefix from 'rest/interceptor/pathPrefix';
import errorCode from 'rest/interceptor/errorCode';
import mime from 'rest/interceptor/mime';

const client = rest.wrap(mime, { mime: 'application/json' })
    .wrap(errorCode, { code: 500 })
    .wrap(pathPrefix, { prefix: `${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}` });

export class UserHeader extends Component {

    requestUserInfo() {
        let token = window.localStorage.getItem('Authorization');
        client({
            method: 'GET',
            path: '/user/',
            headers: {
                Authorization: `Bearer ${token}`
            }
        }).then(data => {
            if (data.entity === 'Unauthorized') {
                window.location.reload();
            } else {
                this.props.setUserInfo(JSON.parse(data.entity));

                this.props.showUserProfile();
            }
        });
    }

    requestUserHistoryBasket() {
        let token = window.localStorage.getItem('Authorization');
        client({
            method: 'GET',
            path: '/basket/users/history',
            headers: {
                Authorization: `Bearer ${token}`
            }
        }).then(data => {
            if (data.entity === 'Unauthorized') {
                window.location.reload();
            } else if (data.entity.length > 0) {
                this.props.showHistoryOfShoppping(data.entity);
                this.props.showUserHistory();
            }
        });
    }

    render() {
        let { userInfo, setUserInState } = this.props;
        return (

            <Row className='d-flex align-items-center justify-content-end mt-1 ml-auto'>
                <Col sm='auto' className='d-flex align-items-center'>
                    <div className='mr-1'>User: {userInfo.username}</div>
                    <Button
                        className='mr-1'
                        variant='light'
                        onClick={() => {
                            this.requestUserInfo();
                        }}>
                        <Link
                            to='/user/profile'
                        >
                            Profile
                        </Link>
                    </Button>
                    <Button
                        className='mr-1'
                        variant='light'
                        onClick={() => {
                            this.requestUserHistoryBasket();
                        }}>
                        <Link
                            to='/user/history'
                        >
                            History
                        </Link>
                    </Button>
                    <Button
                        className='mr-1'
                        variant='light'
                        onClick={() => {
                            window.localStorage.removeItem('Authorization');
                            this.props.showCatalogue();
                            setUserInState({
                                role: 'Guest'
                            });
                        }}>
                        Logout
                    </Button>
                </Col>
            </Row>

        );

    }
}

UserHeader.propTypes = {
    userInfo: PropTypes.object,
    setUserInState: PropTypes.func,
    showHistoryOfShoppping: PropTypes.func,
    showUserHistory: PropTypes.func,
    setUserInfo: PropTypes.func,
    showUserProfile: PropTypes.func,
    showCatalogue: PropTypes.func
};

const mapStoreToProps = function (state) {
    return (
        {
            historyBasket: state.userHeaderReducers.historyBasket
        }
    );
};

export default connect(mapStoreToProps, {
    showHistoryOfShoppping,
    showUserHistory,
    setUserInfo,
    showUserProfile,
    showCatalogue
})(UserHeader);

