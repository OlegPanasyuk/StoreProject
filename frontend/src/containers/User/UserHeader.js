import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Row, Col, Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';

// For Requests to server
import rest from 'rest';
import pathPrefix from 'rest/interceptor/pathPrefix';
import errorCode from 'rest/interceptor/errorCode';
import mime from 'rest/interceptor/mime';

// Redux
import { connect } from 'react-redux';
import { setUserInfo } from '../../actions/actionsUser';
import { showHistoryOfShoppping } from '../../actions/actionsHistoryOfShopping';
import {
    showUserHistory,
    showUserProfile,
    showCatalogue
} from '../../actions/actionsMainContent';

import UserService from '../../Services/UserService';

const client = rest.wrap(mime, { mime: 'application/json' })
    .wrap(errorCode, { code: 500 })
    .wrap(pathPrefix, { prefix: `${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}` });

export class UserHeader extends Component {
    requestUserInfo() {
        const token = window.localStorage.getItem('Authorization');
        const { setUserInfo, showUserProfile } = this.props;
        client({
            method: 'GET',
            path: '/user/',
            headers: {
                Authorization: `Bearer ${token}`
            }
        }).then((data) => {
            if (data.entity === 'Unauthorized') {
                window.location.reload();
            } else {
                setUserInfo(JSON.parse(data.entity));
                showUserProfile();
            }
        });
    }

    requestUserHistoryBasket() {
        const { showHistoryOfShoppping, showUserHistory } = this.props;
        UserService.getUserHistoryOfShopping()
            .then((res) => {
                if (res.data === 'Unauthorized') {
                    window.location.reload();
                } else if (res.data.length > 0) {
                    showHistoryOfShoppping(res.data);
                    showUserHistory();
                }
            });
        // client({
        //     method: 'GET',
        //     path: '/basket/users/history',
        //     headers: {
        //         Authorization: `Bearer ${token}`
        //     }
        // }).then((data) => {
        //     if (data.entity === 'Unauthorized') {
        //         window.location.reload();
        //     } else if (data.entity.length > 0) {
        //         showHistoryOfShoppping(data.entity);
        //         showUserHistory();
        //     }
        // });
    }

    render() {
        const { userInfo, setUserInState, showCatalogue } = this.props;
        return (

            <Row className='d-flex align-items-center justify-content-end mt-1 ml-auto'>
                <Col sm='auto' className='d-flex align-items-center'>
                    <div
                        className='mr-1'
                        style={{
                            color: 'var(--light)'
                        }}
                    >
                        User:
                        {userInfo.username}
                    </div>
                    <Button
                        className='mr-1'
                        variant='light'
                        onClick={() => {
                            this.requestUserInfo();
                        }}
                    >
                        <Link
                            to='/user/profile'
                        >
                            Profile
                        </Link>
                    </Button>
                    <Button
                        className='mr-1'
                        variant='light'
                        onClick={() => {
                            this.requestUserHistoryBasket();
                        }}
                    >
                        <Link
                            to='/user/history'
                        >
                            History
                        </Link>
                    </Button>
                    <Button
                        className='mr-1'
                        variant='light'
                        onClick={() => {
                            window.localStorage.removeItem('Authorization');
                            showCatalogue();
                            setUserInState({
                                role: 'Guest'
                            });
                        }}
                    >
                        Logout
                    </Button>
                </Col>
            </Row>
        );
    }
}

UserHeader.propTypes = {
    userInfo: PropTypes.object.isRequired,
    setUserInState: PropTypes.func.isRequired,
    showHistoryOfShoppping: PropTypes.func.isRequired,
    showUserHistory: PropTypes.func.isRequired,
    setUserInfo: PropTypes.func.isRequired,
    showUserProfile: PropTypes.func.isRequired,
    showCatalogue: PropTypes.func.isRequired
};

const mapStoreToProps = state => (
    {
        historyBasket: state.userHeaderReducers.historyBasket
    }
);

export default connect(mapStoreToProps, {
    showHistoryOfShoppping,
    showUserHistory,
    setUserInfo,
    showUserProfile,
    showCatalogue
})(UserHeader);
