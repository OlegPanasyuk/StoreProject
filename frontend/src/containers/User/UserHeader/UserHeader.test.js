import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import { UserHeader } from '../UserHeader';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    setUserInState: () => { },
    userInfo: {
        username: 'Admin',
        email: 'test@test.com',
        role: 'Guest'
    }
}) {
    const component = shallow(<UserHeader {...props} />);
    return { component, props };
}

describe('UserHeader', () => {
    it('Component must be created', () => {
        const { component } = setupComponent();
        expect(UserHeader.propTypes.userInfo).toBe(PropTypes.object);
        expect(UserHeader.propTypes.setUserInState).toBe(PropTypes.func);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});
