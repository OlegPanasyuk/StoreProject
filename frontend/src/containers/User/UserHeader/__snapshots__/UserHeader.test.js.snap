// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHeader Component must be created 1`] = `
<Bootstrap(Row)
  className="d-flex align-items-center justify-content-end mt-1 ml-auto"
>
  <Col
    as="div"
    className="d-flex align-items-center"
    sm="auto"
  >
    <div
      className="mr-1"
    >
      User: 
      Admin
    </div>
    <Button
      active={false}
      className="mr-1"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      <Link
        to="/user/profile"
      >
        Profile
      </Link>
    </Button>
    <Button
      active={false}
      className="mr-1"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      <Link
        to="/user/history"
      >
        History
      </Link>
    </Button>
    <Button
      active={false}
      className="mr-1"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      Logout
    </Button>
  </Col>
</Bootstrap(Row)>
`;
